### PR TITLE
enh(openid): handle new mechanisms for Open ID Authorization.

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to add new mechanism to define Open ID Authorization. It add the possibility to add Authentication Conditions, rework the acl mapping system and add a contactgroup mapping.

**Fixes** # MON-14870

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by Thebarda — DISMISSED** [review]
OK for the frontend part

**Review by dmyios — DISMISSED** [review]

**Review by jdelpierre — DISMISSED** [review]

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -17037,3 +17037,24 @@ msgstr "Supprimer des collecteurs"
...
+msgstr "URL d'endpoint spécifié invalide"
+
+msgid "Request for authentication conditions custom endpoint has failed"
+msgstr "La demande de conditions d'authentification de l'endpoint personnalisé a échoué"
+
+msgid "Authorized conditions not found in provider conditions"
+msgstr "Les conditions autorisées ne figurent pas dans les conditions du provider"
+
+msgid "The request for roles mapping on custom endpoint has failed"
+msgstr "La demande d'association de rôles a échoué en interrogeant l'url personalisée"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "La demande d'association de rôles a échoué en interrogeant l'URL personnalisée"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -17037,3 +17037,24 @@ msgstr "Supprimer des collecteurs"
 
 msgid "Deploy configuration"
 msgstr "Déployer la configuration"
+
+msgid "Invalid endpoint type specified"
+msgstr "Type d'endpoint spécifié invalide"
+
+msgid "Invalid endpoint URL specified"
+msgstr "URL d'endpoint spécifié invalide"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "URL d'endpoint spécifiée invalide"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -17037,3 +17037,24 @@ msgstr "Supprimer des collecteurs"
...
+msgstr "Les conditions autorisées ne figurent pas dans les conditions du provider"
+
+msgid "The request for roles mapping on custom endpoint has failed"
+msgstr "La demande d'association de rôles a échoué en interrogeant l'url personalisée"
+
+msgid "Invalid roles mapping fetched from provider"
+msgstr "Les roles récupérés depuis le serveur d'identité sont incorrects"
+
+msgid "Role mapping conditions do not match"
+msgstr "Aucune correspondance trouvée pour le couple roles / groupe d'accès"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Aucune correspondance trouvée pour le couple rôles/groupe d'accès"
```

File: src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 4]
```diff
@@ -820,4 +897,468 @@ private function logExceptionInLoginLogFile(string $message, \Exception $e): voi
...
+     * @param array<string,mixed> $conditions
+     * @param AuthenticationConditions $authenticationConditions
+     * @throws AuthenticationConditionsException
+     */
+    private function validateAuthenticationConditions(
+        array $conditions,
+        AuthenticationConditions $authenticationConditions
+    ): void {
+        $authenticationAttributePath = explode(".", $authenticationConditions->getAttributePath());
+        $this->logAuthenticationInfo("Configured Attribute path found", $authenticationAttributePath);
```
Comment (cg-tw) [inline src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]:
```suggestion
        $this->logAuthenticationInfo("Configured attribute path found", $authenticationAttributePath);
```

File: src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 5]
```diff
@@ -820,4 +897,468 @@ private function logExceptionInLoginLogFile(string $message, \Exception $e): voi
...
+        $this->logAuthenticationInfo("Configured Authorized values", $authenticationConditions->getAuthorizedValues());
```
Comment (cg-tw) [inline src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]:
```suggestion
        $this->logAuthenticationInfo("Configured authorized values", $authenticationConditions->getAuthorizedValues());
```

File: src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 6]
```diff
@@ -820,4 +897,468 @@ private function logExceptionInLoginLogFile(string $message, \Exception $e): voi
...
+     * @param array<mixed> $providerAuthenticationConditions
+     * @param string[] $configuredAuthorizedValues
+     * @throws AuthenticationConditionsException
+     */
+    private function validateAuthenticationAttributeOrFail(
+        array $providerAuthenticationConditions,
+        array $configuredAuthorizedValues
+    ): void {
+        if (array_is_list($providerAuthenticationConditions) === false) {
+            $errorMessage = "Invalid Authentication conditions format, array of strings expected";
```
Comment (cg-tw) [inline src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]:
```suggestion
            $errorMessage = "Invalid authentication conditions format, array of strings expected";
```

File: src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 7]
```diff
@@ -820,4 +897,468 @@ private function logExceptionInLoginLogFile(string $message, \Exception $e): voi
...
+     * @return array<string,mixed>
+     * @throws ClientExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws SSOAuthenticationException
+     * @throws ServerExceptionInterface
+     * @throws TransportExceptionInterface
+     */
+    private function sendRequestForCustomAclConditionEndpoint(string $customEndpoint): array
+    {
+        $this->info('Send Request for authentication conditions...');
```
Comment (cg-tw) [inline src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]:
```suggestion
        $this->info('Sending request for authentication conditions...');
```

File: src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 8]
```diff
@@ -820,4 +897,468 @@ private function logExceptionInLoginLogFile(string $message, \Exception $e): voi
...
+     * @param array<mixed> $providerGroupsMapping
+     * @param ContactGroupRelation[] $contactGroupRelations
+     * @throws AuthenticationConditionsException
+     */
+    private function validateGroupsMappingAttributeOrFail(
+        array $providerGroupsMapping,
+        array $contactGroupRelations
+    ): void {
+        if (array_is_list($providerGroupsMapping) === false) {
+            $errorMessage = "Invalid Authentication conditions format, array of strings expected";
```
Comment (cg-tw) [inline src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]:
```suggestion
            $errorMessage = "Invalid authentication conditions format, array of strings expected";
```

File: src/Core/Security/Authentication/Infrastructure/Provider/AclUpdater.php [hunk 9]
```diff
@@ -106,16 +114,22 @@ private function updateAccessGroupsForUser(ContactInterface $user, array $userAc
     private function updateContactGroupsForUser(ContactInterface $user): void
     {
         /** @phpstan-ignore-next-line */
-        $contactGroup = $this->provider->getConfiguration()->getCustomConfiguration()->getContactGroup();
+        $contactGroups = $this->provider->getUserContactGroups();
 
         try {
             $this->info('Updating User Contact Group', [
```
Comment (cg-tw) [inline src/Core/Security/Authentication/Infrastructure/Provider/AclUpdater.php]:
```suggestion
            $this->info('Updating user contact group', [
```

File: src/Core/Security/Authentication/Infrastructure/Provider/OpenId.php [hunk 10]
```diff
@@ -272,11 +239,12 @@ public function getUserAccessGroupsFromClaims(array $claims): array
         $userAccessGroups = [];
         /** @var CustomConfiguration $customConfiguration */
         $customConfiguration = $this->provider->getConfiguration()->getCustomConfiguration();
-        foreach ($customConfiguration->getAuthorizationRules() as $authorizationRule) {
-            if (!in_array($authorizationRule->getClaimValue(), $claims)) {
+        foreach ($customConfiguration->getACLConditions()->getRelations() as $authorizationRule) {
+            $claimValue = $authorizationRule->getClaimValue();
+            if (!in_array($claimValue, $claims)) {
                 $this->info(
                     "Configured Claim Value not found in user claims",
```
Comment (cg-tw) [inline src/Core/Security/Authentication/Infrastructure/Provider/OpenId.php]:
```suggestion
                    "Configured claim value not found in user claims",
```

File: src/Core/Security/ProviderConfiguration/Application/OpenId/UseCase/UpdateOpenIdConfiguration/UpdateOpenIdConfiguration.php [hunk 11]
```diff
@@ -263,4 +268,145 @@ private function updateConfiguration(Configuration $configuration): void
...
+     * @param ContactGroup[] $foundContactGroups
+     */
+    private function logNonExistentContactGroupsIds(array $contactGroupIds, array $foundContactGroups): void
+    {
+        $foundContactGroupsId = [];
+        foreach ($foundContactGroups as $foundAccessGroup) {
+            $foundContactGroupsId[] = $foundAccessGroup->getId();
+        }
+        $nonExistentAccessGroupsIds = array_diff($contactGroupIds, $foundContactGroupsId);
+        $this->error("Access Groups not found", [
```
Comment (cg-tw) [inline src/Core/Security/ProviderConfiguration/Application/OpenId/UseCase/UpdateOpenIdConfiguration/UpdateOpenIdConfiguration.php]:
```suggestion
        $this->error("Access groups not found", [
```

File: src/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php [hunk 12]
```diff
@@ -68,12 +72,19 @@ public function updateConfiguration(Configuration $configuration): void
...
+        $authorizationRules = $customConfiguration->getACLConditions()->getRelations();
 
         $this->info('Removing existing Authorization Rules');
         $this->deleteAuthorizationRules();
         $this->info('Inserting new Authorization Rules');
         $this->insertAuthorizationRules($authorizationRules);
+
+        $contactGroupRelations = $customConfiguration->getGroupsMapping()->getContactGroupRelations();
+
+        $this->info('Removing existing Group Mappings');
```
Comment (cg-tw) [inline src/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php]:
```suggestion
        $this->info('Removing existing group mappings');
```

File: src/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php [hunk 13]
```diff
@@ -68,12 +72,19 @@ public function updateConfiguration(Configuration $configuration): void
 
         /** @var CustomConfiguration $customConfiguration */
         $customConfiguration = $configuration->getCustomConfiguration();
-        $authorizationRules = $customConfiguration->getAuthorizationRules();
+        $authorizationRules = $customConfiguration->getACLConditions()->getRelations();
 
         $this->info('Removing existing Authorization Rules');
```
Comment (cg-tw) [inline src/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php]:
```suggestion
        $this->info('Removing existing authorization rules');
```

File: src/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php [hunk 14]
```diff
@@ -68,12 +72,19 @@ public function updateConfiguration(Configuration $configuration): void
...
         $this->deleteAuthorizationRules();
         $this->info('Inserting new Authorization Rules');
```
Comment (cg-tw) [inline src/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php]:
```suggestion
        $this->info('Inserting new authorization rules');
```

File: src/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php [hunk 15]
```diff
@@ -68,12 +72,19 @@ public function updateConfiguration(Configuration $configuration): void
...
         $this->insertAuthorizationRules($authorizationRules);
+
+        $contactGroupRelations = $customConfiguration->getGroupsMapping()->getContactGroupRelations();
+
+        $this->info('Removing existing Group Mappings');
+        $this->deleteContactGroupRelations();
+        $this->info('Inserting new Group Mappings');
```
Comment (cg-tw) [inline src/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php]:
```suggestion
        $this->info('Inserting new group mappings');
```

File: tests/php/Core/Security/Authentication/Application/UseCase/LoginSession/LoginSessionTest.php [hunk 16]
```diff
@@ -22,524 +22,408 @@
...
+        $this->writeSessionTokenRepository,
+        $this->aclUpdater,
+        $this->menuService,
+        $this->defaultRedirectUri
+    );
+
+    $this->authenticationRequest = LoginRequest::createForLocal("admin", "password", '127.0.0.1');
+});
+
+it('should present an error Response when the Provider configuration is not found', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/Authentication/Application/UseCase/LoginSession/LoginSessionTest.php]:
```suggestion
it('should present an error response when the provider configuration is not found', function () {
```

File: tests/php/Core/Security/Authentication/Application/UseCase/LoginSession/LoginSessionTest.php [hunk 17]
```diff
@@ -22,524 +22,408 @@
...
+        ->method('create')
+        ->with(Provider::LOCAL)
+        ->willThrowException(ProviderException::providerConfigurationNotFound(Provider::LOCAL));
+
+    $useCase($this->authenticationRequest, $presenter);
+    expect($presenter->getResponseStatus())->toBeInstanceOf(ErrorResponse::class);
+});
+
+
+it('should present an UnauthorizedResponse when the authentication fail', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/Authentication/Application/UseCase/LoginSession/LoginSessionTest.php]:
```suggestion
it('should present an UnauthorizedResponse when the authentication fails', function () {
```

File: tests/php/Core/Security/Authentication/Application/UseCase/LoginSession/LoginSessionTest.php [hunk 18]
```diff
@@ -22,524 +22,408 @@
...
+        ->expects($this->once())
+        ->method('create')
+        ->with(Provider::LOCAL)
+        ->willReturn($this->provider);
+
+    $useCase($this->authenticationRequest, $presenter);
+    expect($presenter->getResponseStatus())->toBeInstanceOf(PasswordExpiredResponse::class);
+});
+
+it('should present an UnauthorizedResponse when User is not authorize to login', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/Authentication/Application/UseCase/LoginSession/LoginSessionTest.php]:
```suggestion
it('should present an UnauthorizedResponse when user is not authorized to log in', function () {
```

File: tests/php/Core/Security/Authentication/Application/UseCase/LoginSession/LoginSessionTest.php [hunk 19]
```diff
@@ -22,524 +22,408 @@
...
+        $this->menuService,
+        $this->defaultRedirectUri
+    );
+    $presenter = new LoginPresenterStub();
+    $useCase($this->authenticationRequest, $presenter);
+    expect($presenter->getResponseStatus())->toBeInstanceOf(UnauthorizedResponse::class);
+});
+
+
+it("should present an UnauthorizedResponse when User doesn't exist", function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/Authentication/Application/UseCase/LoginSession/LoginSessionTest.php]:
```suggestion
it("should present an UnauthorizedResponse when user doesn't exist", function () {
```

File: tests/php/Core/Security/Authentication/Application/UseCase/LoginSession/LoginSessionTest.php [hunk 20]
```diff
@@ -22,524 +22,408 @@
...
+    $this->provider
+        ->expects($this->once())
+        ->method('findUserOrFail')
+        ->willThrowException(LegacyAuthenticationException::userNotFound());
+
+    $useCase($this->authenticationRequest, $presenter);
+    expect($presenter->getResponseStatus())->toBeInstanceOf(UnauthorizedResponse::class);
+});
+
+it('should create an user when auto import is enabled', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/Authentication/Application/UseCase/LoginSession/LoginSessionTest.php]:
```suggestion
it('should create a user when auto import is enabled', function () {
```

File: tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+beforeEach(function () {
+    $this->custom_relative_url = '/info';
+    $this->custom_url = 'https://domain.com/info';
+});
+
+it('should throw an exception with a bad endpoint type', function () {
+    (new Endpoint('bad_type', $this->custom_relative_url));
+})->throws(InvalidEndpointException::class, InvalidEndpointException::invalidType()->getMessage());
+
+it('should throw an exception with a bad relative url', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php]:
```suggestion
it('should throw an exception with a bad relative URL', function () {
```

File: tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+    (new Endpoint(Endpoint::CUSTOM, 'bad_relative_url'));
+})->throws(InvalidEndpointException::class, InvalidEndpointException::invalidUrl()->getMessage());
+
+it('should return an EndpointCondition instance with a correct relative url', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php]:
```suggestion
it('should return an EndpointCondition instance with a correct relative URL', function () {
```

File: tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+    $endpointCondition = new Endpoint(Endpoint::CUSTOM, $this->custom_relative_url);
+    expect($endpointCondition->getUrl())->toBe($this->custom_relative_url);
+});
+
+it('should return an EndpointCondition instance with a correct url', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php]:
```suggestion
it('should return an EndpointCondition instance with a correct URL', function () {
```

File: tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+    $endpointCondition = new Endpoint(Endpoint::CUSTOM, $this->custom_url);
+    expect($endpointCondition->getUrl())->toBe($this->custom_url);
+});
+
+it('should return an EndpointCondition instance with an empty url if type is not custom', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php]:
```suggestion
it('should return an EndpointCondition instance with an empty URL if type is not custom', function () {
```

File: tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php [hunk 25]
```diff
@@ -0,0 +1,75 @@
...
+    $endpointCondition = new Endpoint(Endpoint::INTROSPECTION, $this->custom_url);
+    expect($endpointCondition->getUrl())->toBeNull();
+
+    $endpointCondition = new Endpoint(Endpoint::USER_INFORMATION, $this->custom_url);
+    expect($endpointCondition->getUrl())->toBeNull();
+});
+
+it('should throw an exception with a null url and a custom type', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php]:
```suggestion
it('should throw an exception with a null URL and a custom type', function () {
```

File: tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php [hunk 26]
```diff
@@ -0,0 +1,75 @@
...
+    (new Endpoint(Endpoint::CUSTOM, null));
+})->throws(InvalidEndpointException::class, InvalidEndpointException::invalidUrl()->getMessage());
+
+it('should throw an exception with a empty url and a custom type', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/ProviderConfiguration/Domain/Model/EndpointTest.php]:
```suggestion
it('should throw an exception with an empty URL and a custom type', function () {
```

File: www/front_src/src/Authentication/Openid/index.test.tsx [hunk 27]
```diff
@@ -324,6 +339,10 @@ describe('Openid configuration form', () => {
   it('saves the openid configuration when a field is modified and the "Save" button is clicked', async () => {
```
Comment (cg-tw) [inline www/front_src/src/Authentication/Openid/index.test.tsx]:
```suggestion
  it('saves the OpenID configuration when a field is modified and the "Save" button is clicked', async () => {
```

File: www/front_src/src/Authentication/Openid/index.test.tsx [hunk 28]
```diff
@@ -506,23 +539,121 @@ describe('Openid configuration form', () => {
...
+    await waitFor(() => {
+      expect(screen.getAllByLabelText(labelDefineYourEndpoint)).toHaveLength(2);
+    });
+
+    userEvent.click(screen.getAllByLabelText(labelOther)[0]);
+
+    expect(screen.getAllByLabelText(labelDefineYourEndpoint)).toHaveLength(3);
+  });
+
+  it('displays new input field if a user type a text in the latest element of the field "condition value" ', async () => {
```
Comment (cg-tw) [inline www/front_src/src/Authentication/Openid/index.test.tsx]:
```suggestion
  it('displays a new input field if a user types a text in the latest element of the field "condition value" ', async () => {
```

File: www/front_src/src/Authentication/Openid/index.test.tsx [hunk 29]
```diff
@@ -506,23 +539,121 @@ describe('Openid configuration form', () => {
...
+
+    userEvent.type(
+      screen.getAllByLabelText(labelConditionValue)[1],
+      'some text',
+    );
+
+    expect(screen.getAllByLabelText(labelConditionValue)).toHaveLength(3);
+  });
+
+  it('displays a new delete icon if a user type a text in the latest element of the field "condition value" ', async () => {
```
Comment (cg-tw) [inline www/front_src/src/Authentication/Openid/index.test.tsx]:
```suggestion
  it('displays a new Delete icon if a user types a text in the latest element of the field "condition value" ', async () => {
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 30]
```diff
@@ -17037,3 +17037,24 @@ msgstr "Supprimer des collecteurs"
...
 msgstr "Déployer la configuration"
+
+msgid "Invalid endpoint type specified"
+msgstr "Type d'endpoint spécifié invalide"
+
+msgid "Invalid endpoint URL specified"
+msgstr "URL d'endpoint spécifié invalide"
+
+msgid "Request for authentication conditions custom endpoint has failed"
+msgstr "La demande de conditions d'authentification de l'endpoint personnalisé a échoué"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "La demande d'endpoint personnalisé  de conditions d'authentification a échoué"
```

Comment (Thebarda) [reply lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]: @cg-tw There is a double space in your suggestion

**Review by callapa — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 31]
```diff
@@ -17037,3 +17037,24 @@ msgstr "Supprimer des collecteurs"
...
 msgstr "Déployer la configuration"
+
+msgid "Invalid endpoint type specified"
+msgstr "Type d'endpoint spécifié invalide"
+
+msgid "Invalid endpoint URL specified"
+msgstr "URL d'endpoint spécifiée invalide"
+
+msgid "Request for authentication conditions custom endpoint has failed"
+msgstr "La demande d'endpoint personnalisé  de conditions d'authentification a échoué"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17048]:
```suggestion
msgstr "La demande d'endpoint personnalisé de conditions d'authentification a échoué"
```

**Review by cg-tw — APPROVED** [review]

**Pull request merged**